sphinx/testing/_matcher/cleaner.py

```diff
@@ -0,0 +1,219 @@
+from __future__ import annotations
+
+__all__ = ()
+
+import itertools
+from functools import reduce
+from itertools import filterfalse
+from typing import TYPE_CHECKING
+
+from sphinx.testing._matcher import engine, util
+from sphinx.testing._matcher.options import get_option
+from sphinx.util.console import strip_colors, strip_control_sequences
+
+if TYPE_CHECKING:
+    import re
+    from collections.abc import Iterable, Sequence
+    from typing import TypeVar
+
+    from typing_extensions import Unpack
+
+    from sphinx.testing._matcher.options import (
+        DeletePattern,
+        Flavor,
+        LinePredicate,
+        Options,
+        StripChars,
+    )
+
+    _StrIterableT = TypeVar('_StrIterableT', bound=Iterable[str])
+
+
+def clean_text(text: str, /, **options: Unpack[Options]) -> Iterable[str]:
+    """Clean a text."""
+    ctrl = get_option(options, 'ctrl')
+    color = get_option(options, 'color')
+    text = strip_ansi(text, ctrl=ctrl, color=color)
+
+    text = strip_chars(text, get_option(options, 'strip'))
+    lines = text.splitlines(get_option(options, 'keepends'))
+    return clean_lines(lines, **options)
+
+
+def clean_lines(lines: Iterable[str], /, **options: Unpack[Options]) -> Iterable[str]:
+    """Clean a list of lines."""
+    lines = strip_lines(lines, get_option(options, 'stripline'))
+    # Removing empty lines first ensures that serial duplicates can
+    # be eliminated in one cycle. Inverting the order of operations
+    # is not possible since empty lines may 'hide' duplicated lines.
+    empty = get_option(options, 'empty')
+    compress = get_option(options, 'compress')
+    unique = get_option(options, 'unique')
+    lines = filter_lines(lines, empty=empty, compress=compress, unique=unique)
+
+    delete = get_option(options, 'delete')
+    flavor = get_option(options, 'flavor')
+    lines = prune(lines, delete, flavor=flavor)
+
+    return ignore_lines(lines, get_option(options, 'ignore'))
+
+
+def strip_ansi(text: str, /, ctrl: bool = False, color: bool = False) -> str:
+    """Strip ANSI escape sequences.
+
+    :param text: The text to clean.
+    :param ctrl: If true, remove non-color ANSI escape sequences.
+    :param color: If true, remove color ANSI escape sequences.
+    :return: The cleaned text.
+    """
+    # non-color control sequences must be stripped before colors
+    text = text if ctrl else strip_control_sequences(text)
+    text = text if color else strip_colors(text)
+    return text
+
+
+def strip_chars(text: str, chars: StripChars = True, /) -> str:
+    """Strip expected characters from *text*."""
+    if isinstance(chars, bool):
+        return text.strip() if chars else text
+
+    if isinstance(chars, str) or chars is None:
+        return text.strip(chars)
+
+    msg = 'expecting a boolean, a string or None for %r, got: %r' % ('strip', chars)
+    raise ValueError(msg)
```

Comment:
```suggestion
    return text.strip(chars)
```

I think I can let Python handle the exception itself.